kotlin-eclipse-ui/src/org/jetbrains/kotlin/ui/editors/DiagnosticAnnotationUtil.java

```diff
@@ -177,9 +178,14 @@ public void updateAnnotations(
             @NotNull Map<IFile, List<DiagnosticAnnotation>> annotations,
             @NotNull Predicate<Annotation> replacementAnnotationsPredicate) {
         try {
+            List<DiagnosticAnnotation> newAnnotations = null;
             IFile file = EditorUtil.getFile(editor);
+            if (file != null) {
+                newAnnotations = annotations.get(file);
+            } else {
+                KotlinLogger.log(IStatus.ERROR, "Failed to retrieve IFile from editor " + editor, null);
+            }
```

Comment:
This part of code could be improved a little:

```java
List<DiagnosticAnnotation> newAnnotations;
if (file != null && annotations.contains(file)) {
  ...
} else {
  ...
  newAnnotaions = Collections.emptyList();
}
```

As a result newAnnotations will be nonnull